### PR TITLE
chore: refactor Attachment to be functional component

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

This PR converts `Attachment` component to be functional. As in all other cases it contains a breaking change with restricted props set that will be passed to styles functions.

## Prop sets

| `Attachment`    |
| --------- |
| `actionable` |
| `disabled` |

***

## Notable changes

- Adds a test for telemetry capturing
- `progress` was moved out from CSS styles as in `Slider` component

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12576)

### CODE REVIEW

File: packages/fluentui/react-northstar/test/specs/commonTests/isConformant.tsx [hunk 1]
```diff
@@ -554,32 +555,25 @@ export default function isConformant(
     });
   });
 
-  const validListenerNames = _.reduce(syntheticEvent.types, (result, { listeners }) => [...result, ...listeners], []);
-
   // ---------------------------------------
-  // Opt-in tests
```
Comment (layershifter) [inline packages/fluentui/react-northstar/test/specs/commonTests/isConformant.tsx:560]: This was never used, IDK what it does and why 🤔

File: packages/fluentui/react-northstar/src/themes/teams/components/Attachment/attachmentStyles.ts [hunk 2]
```diff
@@ -130,7 +131,6 @@ const attachmentStyles: ComponentSlotStylesPrepared<AttachmentProps, AttachmentV
     display: 'block',
     bottom: 0,
     left: 0,
-    width: `${p.progress}%`,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Attachment/attachmentStyles.ts:133]: I moved this to `style` prop on `progress` as in `Slider` component as obviously we don't want to have one hundled entries in cache to different values of `progress`

File: packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx [hunk 3]
```diff
@@ -49,103 +53,130 @@ export interface AttachmentProps extends UIComponentProps, ChildrenComponentProp
...
+          {Text.create(header, {
+            defaultProps: () => ({ styles: resolvedStyles.header }),
+          })}
+
+          {Text.create(description, {
+            defaultProps: () => ({ styles: resolvedStyles.description }),
+          })}
+        </div>
+      )}
+      {Button.create(action, {
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx:139]: Please why don't you have check that `action` is defined, compared with the previous code?

Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx:139]: Because it will be done by `.create()` factory 😺

**Pull request merged**